input/system/selfhosted/otel_handler.go

```diff
@@ -0,0 +1,210 @@
+		}
+		if rv.Key == "user_name" {
+			logLine.Username = rv.Value.GetStringValue()
+		}
+		if rv.Key == "database_name" {
+			logLine.Database = rv.Value.GetStringValue()
+		}
+		if rv.Key == "process_id" {
+			backendPid, _ := strconv.ParseInt(rv.Value.GetStringValue(), 10, 32)
+			logLine.BackendPid = int32(backendPid)
+		}
+		if rv.Key == "application_name" {
+			logLine.Application = rv.Value.GetStringValue()
+		}
+		if rv.Key == "session_line_num" {
+			logLineNumber, _ := strconv.ParseInt(rv.Value.GetStringValue(), 10, 32)
+			logLine.LogLineNumber = int32(logLineNumber)
+		}
+		if rv.Key == "message" {
+			logLine.Content = rv.Value.GetStringValue()
+		}
+		if rv.Key == "detail" {
+			detailLineContent = rv.Value.GetStringValue()
+		}
+		if rv.Key == "error_severity" {
+			logLine.LogLevel = pganalyze_collector.LogLineInformation_LogLevel(pganalyze_collector.LogLineInformation_LogLevel_value[rv.Value.GetStringValue()])
+		}
+	}
+	if detailLineContent != "" {
+		detailLine := logLine
+		detailLine.Content = detailLineContent
+		detailLine.LogLevel = pganalyze_collector.LogLineInformation_DETAIL
+		return logLine, &detailLine
+	}
+
+	return logLine, nil
+}
+
+func skipDueToK8sFilter(kubernetes *common.KeyValueList, server *state.Server, prefixedLogger *util.Logger) bool {
+	var k8sPodName string
+	var k8sNamespaceName string
+
+	k8sLabels := make(map[string]string)
+	for _, rv := range kubernetes.Values {
+		if rv.Key == "pod_name" {
+			k8sPodName = rv.Value.GetStringValue()
+		}
+		if rv.Key == "namespace_name" {
+			k8sNamespaceName = rv.Value.GetStringValue()
+		}
+		if rv.Key == "labels" {
+			for _, ll := range rv.Value.GetKvlistValue().Values {
+				k8sLabels[ll.Key] = ll.Value.GetStringValue()
+			}
+		}
+	}
+
+	if server.Config.LogOtelK8SPod != "" {
+		parts := strings.SplitN(server.Config.LogOtelK8SPod, "/", 2)
+		if len(parts) == 2 {
+			if k8sNamespaceName != parts[0] || k8sPodName != parts[1] {
+				return true
+			}
+		} else if len(parts) == 1 {
+			if k8sPodName != parts[0] {
+				return true
+			}
+		} else {
+			prefixedLogger.PrintWarning("Pod specification for OTel server not valid (need zero or one / separator): \"%s\", skipping log record\n", server.Config.LogOtelK8SPod)
+			return true
+		}
+	}
+
+	if server.Config.LogOtelK8SLabels != "" {
+		selectors := strings.Split(server.Config.LogOtelK8SLabels, ",")
+		for _, selector := range selectors {
+			parts := k8sSelectorRegexp.FindStringSubmatch(selector)
+			if parts != nil {
+				selKey := parts[1]
+				selEq := parts[2] == "=" || parts[2] == "=="
+				selNotEq := parts[2] == "!="
+				selValue := parts[3]
+				v, ok := k8sLabels[selKey]
+				if ok {
+					if (selEq && v != selValue) || (selNotEq && v == selValue) {
+						return true
+					}
+				}
+			} else {
+				prefixedLogger.PrintWarning("Label selector for OTel server not valid: \"%s\", skipping log record\n", server.Config.LogOtelK8SLabels)
+				return true
+			}
+		}
+	}
```

Comment:
Similar to above, I think it'd be better parse this into a structured form when reading the config, and make it easier to work with for consumers.